qiskit/aqua/operators/gradients/circuit_qfis/lin_comb_full.py

```diff
@@ -0,0 +1,386 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The module for Quantum the Fisher Information."""
+
+import copy
+from copy import deepcopy
+from typing import List, Union, Optional, Tuple
+
+import numpy as np
+from qiskit.aqua import AquaError
+from qiskit.aqua.operators import ListOp
+from qiskit.aqua.operators.operator_globals import I, Z, Y, X
+from qiskit.aqua.operators.state_fns import StateFn, CircuitStateFn
+from qiskit.circuit import Gate, Qubit
+from qiskit.circuit import (QuantumCircuit, QuantumRegister, ParameterVector,
+                            ParameterExpression)
+from qiskit.circuit.library import RZGate, RXGate, HGate, XGate, SdgGate, SGate, ZGate, UGate
+
+from .circuit_qfi import CircuitQFI
+from ..circuit_gradients.lin_comb import LinComb
+from ..derivative_base import DerivativeBase
+
+
+class LinCombFull(CircuitQFI):
+    r"""Compute the full Quantum Fisher Information (QFI) given a pure, parametrized quantum state
+    with an additional working qubit and the linear combination of unitaries approach.
+
+    The QFI is:
+
+        [QFI]kl= Re[〈∂kψ|∂lψ〉−〈∂kψ|ψ〉〈ψ|∂lψ〉] * 4.
+    """
+
+    def convert(self,
+                operator: CircuitStateFn,
+                params: Optional[Union[ParameterExpression, ParameterVector,
+                                       List[ParameterExpression]]] = None,
+                ) -> ListOp:
+        r"""
+        Args:
+            operator: The operator corresponding to the quantum state |ψ(ω)〉for which we compute
+                the QFI
+            params: The parameters we are computing the QFI wrt: ω
+
+        Returns:
+            ListOp[ListOp] where the operator at position k,l corresponds to QFI_kl
+
+        Raises:
+            AquaError: If one of the circuits could not be constructed.
+            TypeError: If ``operator`` is an unsupported type.
+        """
+
+        # QFI & phase fix observable
+        qfi_observable = ~StateFn(4 * Z ^ (I ^ operator.num_qubits))
+        phase_fix_observable = ~StateFn((X + 1j * Y) ^ (I ^ operator.num_qubits))
+        # see https://arxiv.org/pdf/quant-ph/0108146.pdf
+
+        if isinstance(operator, CircuitStateFn):
+            pass
+        else:
+            raise TypeError(
+                'The gradient framework is compatible with states that are given as CircuitStateFn')
+
+        if not isinstance(params, (list, np.ndarray)):
+            params = [params]
+        state_qc = copy.deepcopy(operator.primitive)
```

Comment:
As the opflow is immutable, `state_qc` will not be changed on the operator side. Do we change it here in the code? If not we don't need to copy it

Reply:
I'll check

Reply:
seems to be fine